### PR TITLE
SPCTR-3593 Resolved Blog home page style broken issue.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-3593

### Screenshots

- https://d.pr/v/36EcE5

### Types of changes

- Bug fix (non-breaking change which fixes an issue) 

### How has this been tested?

- Verify it in screencast added.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-post-assets.php [hunk 1]
```diff
@@ -305,29 +305,40 @@ private function determine_template_post_type( $post_id ) {
 			'is_category'   => 'category',
 			'is_date'       => 'date',
 			'is_embed'      => 'embed',
-			'is_front_page' => 'front-page',
+			'is_front_page' => 'home',
 			'is_home'       => 'home',
```
Comment (adi3890) [inline classes/class-uagb-post-assets.php:309]: do we need them here if we are testing it separately before the loop

File: classes/class-uagb-post-assets.php [hunk 2]
```diff
@@ -305,29 +305,40 @@ private function determine_template_post_type( $post_id ) {
...
 				} elseif ( is_singular() ) { // Applies to single post of any post type ( attachment, page, custom post types).
-					return 'single-' . $object->post_type . '-' . $object->post_name;
+					$name_decoded   = urldecode( $object->post_name );
+					$template_types = get_block_templates();
+					foreach ( $template_types as $template_type ) {
+						if ( 'single-' . $object->post_type === $template_type->slug ) {
+							$what_post_type = 'single-' . $object->post_type;
+						} elseif ( 'single-' . $object->post_type . '-' . $name_decoded === $template_type->slug ) {
+							$what_post_type = 'single-' . $object->post_type . '-' . $name_decoded;
+						}
```
Comment (adi3890) [inline classes/class-uagb-post-assets.php:335]: should we break the loop as sson as it satisfies the condition
inthat case we either break or return $what_post_type.

Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php:335]: @adi3890 I tried that but its breaking the style for single-${post_type}-${post_name}-template.

Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php]: As suggested used array_column to optimize for each and return when condition satisfy

**Review by adi3890 — APPROVED** [review]

**Pull request merged**